### PR TITLE
Add sling-ready grenades and Molotov (#50643)

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
Features "Add sling-ready grenades and Molotov that can be thrown by modified staff sling"

#### Purpose of change
In response to #50643 - add in the ability to throw specially modified grenades with slings

#### Describe the solution
- Modified standard sling back to use pebble as suggested below
- Added sling-ready_grenade ammo type
- Created sling-ready versions of standard grenade and molotov that can be thrown as well as fired from sling
- Added range to both that is Strength 10 throw +5 squares
- Added recipes for creating sling ready grenade and molotov - grenade + fuse
- Created grenade cradle gunmod for staff sling to allow sling to throw grenades, added recipe for same

#### Describe alternatives you've considered
Modify existing grenades to be ammo rather than tools, however due to the short fuses these are much harder to throw with a sling unmodified

#### Testing
Loaded into game and tested, see screenshots below.

#### Additional context
Slings 1

Slings 2

Slings 3

Slings 4

Slings 5

Slings 6

Slings 7

Slings 8

Slings 9

### CODE REVIEW

File: data/json/items/ammo/sling-ready_grenade.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+      "target": "molotov_lit",
+      "msg": "You light the Molotov cocktail!",
+      "target_charges": 1,
+      "active": true,
+      "moves": 75,
+      "need_fire": 1,
+      "menu_text": "Light rag",
+      "type": "transform"
+    },
+    "flags": [ "NPC_ACTIVATE", "NO_REPAIR" ]
```
Comment (LaVeyanFiend) [inline data/json/items/ammo/sling-ready_grenade.json:29]:
```suggestion
    "flags": [ "NPC_ACTIVATE", "NO_REPAIR" ],
```

Comment (occulti) [reply data/json/items/ammo/sling-ready_grenade.json]: Thank you, have corrected in the latest commit.

**Comment by kevingranade** [comment]
The outlined process would not make the grenades sling-ready, it turns them into a type of sling (sometimes called a comet sling).
You basically tie a rope to it, then throw the whole rope-and-grenade assembly.  You could possibly just do the fuze adjustment and throw one with a staff sling, but a regular sling fires much, much smaller projectiles and is not going to work with something this large.

**Comment by Secret-Cisco** [comment]
Don't regular slings and staff slings both throw rocks now?

**Comment by kevingranade** [comment]
They do, but that's wrong and needs fixing.

**Comment by Secret-Cisco** [comment]
I had assumed you would be fine with that since you changed slings to rocks from pebbles. I suppose it probably wouldn't be feasible/reasonable if slings go down a projectile size to let them still throw grenade projectiles at a much decreased range in addition to much faster pebble sized bullets?

As far as I know the only reason thing preventing an ordinary sling from throwing something as large and heavy as a grenade (~400g) is the size of the cradle and the intended velocity of the projectile for range and penetration. Penetration wouldn't be important for a grenade since the explosives and fragments handle that part.

**Comment by kevingranade** [comment]
The size of the cradle is pretty important yes, you simply need a sling designed to hold a larger projectile.

**Comment by occulti** [comment]
@kevingranade I see what you mean entirely here about the sling size - it might be worth then creating a grenade sling and staff sling item and recipe to use their own ammo\rocks and changing the existing slings to use pebbles again?

As far as I could see the purpose of the fuse would be to burn away on launch (the comet sling idea was secondary) to give grenades the necessary fuse length to be launched from a sling - basically a way to justify that the grenades will need to be modified to become ammo in order to work in a sling pocket.

**Comment by occulti** [comment]
Some anecdotes regrading using staff slings to throw grenades here, might not be worth anything but interesting: http://slinging.org/forum/YaBB.pl?num=1353551226/3

Staff slings may be the way to go here as you suggest - if they are able to throw rocks (unlike the smaller sling which could be downgraded to pebbles) then they should be able to throw modified grenades, possibly with a modified cradle.

**Comment by Secret-Cisco** [comment]
If you're going to revert the sling to pebbles it might be worth looking at the commit that made the change https://github.com/CleverRaven/Cataclysm-DDA/commit/3f844cb3587322235dafc41b666d2f4463af9fdb#diff-029525d6c76c6c1c608011958816c8806feb8860df08168262a3a4ab9e29476b

The description was also altered to reflect the ammunition change.

**Pull request merged**